src/Components/Web.JS/src/Platform/Circuits/CircuitManager.ts

```diff
@@ -233,43 +285,139 @@ export class CircuitManager implements DotNet.DotNetCallDispatcher {
     return true;
   }
 
+  public async pause(remote?: boolean): Promise<boolean> {
+    if (!this._circuitId) {
+      this._logger.log(LogLevel.Error, 'Circuit host not initialized.');
+      return false;
+    }
+
+    if (this._connection!.state !== HubConnectionState.Connected) {
+      this._logger.log(LogLevel.Trace, 'Pause can only be triggered on connected circuits.');
+      return false;
+    }
+
+    if (this._resumingState.isInprogress()) {
+      this._logger.log(LogLevel.Trace, 'Circuit is currently resuming...');
+      return false;
+    }
+
+    if (this._pausingState.isInprogress()) {
+      this._logger.log(LogLevel.Trace, 'Waiting for the circuit to finish pausing...');
+      return this._pausingState.currentProgress();
+    }
+
+    if (this._pausingState.lastValue() === true) {
+      // If the circuit is already paused, we don't need to do anything.
+      this._logger.log(LogLevel.Trace, 'Circuit is already paused.');
+      return true;
+    }
+
+    this._pausingState.reset();
+    const pausingPromise = this._pausingState.currentProgress();
+
+    try {
+      this._logger.log(LogLevel.Trace, 'Pausing the circuit...');
+
+      // Notify the reconnection handler that we are pausing the circuit.
+      // This is used to trigger the UI display.
+      this._options.reconnectionHandler?.onConnectionDown(this._options.reconnectionOptions, undefined, true, remote);
```

Comment:
is this behavior "breaking change" ?

Reply:
No, in the sense that is a new feature.

If you have a custom reconnectionHandler and you call `pause()` in your code is up to you to make sure that your handler is capable of handling it.

The default handler (which is by far the most common case) will handle this automatically, same as the built-in UI. If you had a custom UI on previous versions it will behave as the reconnect UI